File: ChainOfResponsibility/ChainOfResponsibilityShop/ChainOfResponsibilityShop/GoldHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChainOfResponsibilityShop
{
    public class GoldHandler : BaseDiscountHandler
    {
        public override void GetDiscount(string clientType)
        {
            if (clientType == "gold")
            {
                Console.WriteLine("Gold client gets 20% discount.");
            }
            else 
            {
            Next(clientType);
            }
        }
    }
}

File: ChainOfResponsibility/JavaExample/ChainOfResponsibility/src/GoldHandler.java
public class GoldHandler extends BaseDiscountHandler{
    @Override
    public void getDiscount(String clientType) {
        if ("gold".equalsIgnoreCase(clientType)) {
            System.out.println("Gold client gets 20% discount.");
        } else {
            next(clientType);
        }
    }
}
